### DISCUSSION TITLE
Unity - Microsoft Datasync SQLiteStore

### DISCUSSION THREAD

**Discussion by darren-beckett** [Azure/azure-mobile-apps#626] [Q&A]
I cannot get Offline Sync working in Unity.

**Microsoft.Datasync.Client.SQLiteStore**
I followed everything here: https://learn.microsoft.com/en-us/azure/developer/mobile-apps/azure-mobile-apps/overview

What the code currently does:
1) It successfully creates the SQLite Database and Offline Table locally
2) It can retrieve  the Table data from .Net Core (Azure Table)
3) It runs all the Sync code (Push/Pull) without error
4) But, no data is actually inserted into the SQLite database

Note: SQLite is working by itself, i can CRUD with it no problem.

Has anyone had Offline Sync working in Unity?

**Comment by adrianhall** [comment]
Unity is not something I have tested.  There is no reason why it shouldn't work, but it's definitely not a tested scenario.

Do you have a test project I can use to diagnose the issue?  If you do, please file an issue (requesting Unity support) and point me to your test project.

**Reply by darren-beckett** [reply]
I have submitted [Issue 631](https://github.com/Azure/azure-mobile-apps/issues/631).

I have created a Unity Package (zip), it has basic scripts to communicate with the TodoAppService.Net6 sample.
A Unity Scene is included (SQLiteScene), which has 2 GameObjects with SqliteHelper & SqliteTest scripts attached. Just set SQLiteScene in Build Settings.

It is able to create the sqlite db locally and insert records, but will not Push new records back to the server or resync edited remote records.
[Unity-DataSync-SQLite.zip](https://github.com/Azure/azure-mobile-apps/files/10812521/Unity-DataSync-SQLite.zip)
